### PR TITLE
T-Proxy for peering tests

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Support t-proxy in acceptance tests for peers

How I've tested this PR:
- Ran acceptance tests locally

How I expect reviewers to test this PR:
- 👀 

Checklist:
- [x] Tests added

### CODE REVIEW

File: acceptance/tests/peering/peering_connect_namespaces_test.go [hunk 1]
```diff
@@ -79,16 +76,18 @@ func TestPeering_ConnectNamespaces(t *testing.T) {
 				"global.peering.enabled":        "true",
 				"global.enableConsulNamespaces": "true",
 
-				"global.image": "hashicorp/consul-enterprise:1.13.0-alpha2-ent",
+				// "global.image": "hashicorp/consul-enterprise:1.13.0-alpha2-ent",
+				"global.image":    "thisisnotashwin/consul@sha256:0733380a1a177d269c53fff62464e3a4840ea0c3ca24c6164282f8a822ec8825",
+				"global.imageK8S": "thisisnotashwin/consul-k8s@sha256:6fe1ec532876073813c824f27b2c972c03a41376e0729a502f6f3302dc352379",
```
Comment (ndhanushkodi) [inline acceptance/tests/peering/peering_connect_namespaces_test.go:81]: Just confirming these are the images we want, and if so we can remove L79!

File: acceptance/tests/peering/peering_connect_test.go [hunk 2]
```diff
@@ -52,21 +49,25 @@ func TestPeering_Connect(t *testing.T) {
 			commonHelmValues := map[string]string{
 				"global.peering.enabled": "true",
 
-				"global.image": "hashicorp/consul:1.13.0-alpha2",
+				// "global.image": "hashicorp/consul:1.13.0-alpha2",
+				"global.image":    "thisisnotashwin/consul@sha256:0733380a1a177d269c53fff62464e3a4840ea0c3ca24c6164282f8a822ec8825",
+				"global.imageK8S": "thisisnotashwin/consul-k8s@sha256:6fe1ec532876073813c824f27b2c972c03a41376e0729a502f6f3302dc352379",
```
Comment (ndhanushkodi) [inline acceptance/tests/peering/peering_connect_test.go:54]: We can remove L52

**Review by ndhanushkodi — APPROVED** [review]
Looks good Ashwin! Thanks for all the debugging!

Should we re-enable tproxy on the nightly tests with this PR or on a follow on PR since we had removed tproxy due to peering not yet supporting tproxy?

**Comment by thisisnotashwin** [comment]
> Should we re-enable tproxy on the nightly tests with this PR or on a follow on PR since we had removed tproxy due to peering not yet supporting tproxy?

Great catch! I just enabled t-proxy on the nightly tests.

File: charts/consul/templates/server-statefulset.yaml [hunk 3]
```diff
@@ -296,6 +296,7 @@ spec:
                 -config-dir=/consul/userconfig/{{ .name }} \
                 {{- end }}
                 {{- end }}
+                -hcl='ports { grpc = 8503 }' \
```
Comment (ishustava) [inline charts/consul/templates/server-statefulset.yaml:299]: could we put it in server configmap?

File: charts/consul/templates/server-service.yaml [hunk 4]
```diff
@@ -39,6 +39,9 @@ spec:
       port: 8501
       targetPort: 8501
     {{- end }}
+    - name: grpc
+      port: 8503
+      targetPort: 8503
```
Comment (ishustava) [inline charts/consul/templates/server-service.yaml:44]: since it's not a host port, could it be 8502?

Comment (thisisnotashwin) [reply charts/consul/templates/server-service.yaml:44]: you mean the port is 8502 but the targetPort remains 8503?

File: charts/consul/templates/server-statefulset.yaml (hunk 3 above)
Comment (thisisnotashwin) [reply charts/consul/templates/server-statefulset.yaml:299]: Good catch!

File: charts/consul/templates/server-service.yaml (hunk 4 above)
Comment (thisisnotashwin) [reply charts/consul/templates/server-service.yaml:44]: Because it will need to be a hostPort on kind. Then it will clash with the client GRPC port.

Comment (ishustava) [reply charts/consul/templates/server-service.yaml:44]: you're totally right, I looked at the service instead of the stateful set!

**Review by ishustava — APPROVED** [review]

**Pull request merged**